sbncode/TPCReco/VertexStub/StubMergeAlgorithms.cxx

```diff
@@ -30,14 +30,11 @@ double sbn::GetEfield(const detinfo::DetectorPropertiesData &dprop, const spacec
   return EField;
 }
 
-geo::Point_t sbn::GetLocationAtWires(const spacecharge::SpaceCharge *sce, geo::Point_t loc, float xsign) { 
+geo::Point_t sbn::GetLocationAtWires(const spacecharge::SpaceCharge *sce, const geo::GeometryCore *geo, geo::Point_t loc, geo::TPCID TPC, float xsign) { 
   if (sce && sce->EnableCalSpatialSCE()) {
-    // fix negative sign in TPC 0
-    int corr = 1;
-    float xx = loc.X();
-    if (xx < 0) { corr = -1; }
+    // Returned X is the drift -- multiply by the drift direction to undo this
+    int corr = geo->TPC(TPC).DriftDir()[0];
```

Comment:
I _suggest_ using the name of the coordinate rather than the index:
```suggestion
    int corr = geo->TPC(TPC).DriftDir().X();
```
It is more expressive, and indices are not supported in `geo::Vector_t` vectors (which in the future might even become the default vector type — i.e. it should but probably won't).